lib/helpers/authenticate.js

```diff
@@ -28,42 +28,45 @@ module.exports = function authenticate(options, req, res, callback) {
       return callback(new Error('Invalid username or password.'));
     }
 
-    var authenticator = new stormpath.OAuthPasswordGrantRequestAuthenticator(application);
-
-    authenticator.authenticate(options, function (err, authResult) {
+    passwordGrant(config, options, function (err, oauthAccessTokenResult) {
       if (err) {
         logger.info('Error when trying to authenticate user.');
         return callback(err);
       }
 
-      authResult.getAccount(function (err, account) {
+      var issuer = config.okta.org + '/oauth2/' + config.okta.authorizationServerId;
+
+      var accessTokenAuthenticator = new AccessTokenAuthenticator().forIssuer(issuer).withLocalValidation();
+
+      accessTokenAuthenticator.authenticate(oauthAccessTokenResult.access_token, function (err, jwt) {
+
         if (err) {
-          logger.info('Error when trying to retrieve the account for the authenticated user.');
+          logger.info(err);
           return callback(err);
         }
 
-        expandAccount(account, config.expand, logger, function (err, expandedAccount) {
+        getAppUserById(config.okta.org, config.okta.appId, config.okta.apiToken, jwt.body.uid, (err, user) => {
           if (err) {
-            logger.info('Error when trying to expand the account of the authenticated user.');
+            logger.info(err);
             return callback(err);
           }
 
-          req.user = expandedAccount;
-          createSession(authResult, expandedAccount, req, res);
+          createSession(oauthAccessTokenResult, user, req, res);
 
           if (config.postLoginHandler) {
-            return config.postLoginHandler(expandedAccount, req, res, function (err) {
+            return config.postLoginHandler(user, req, res, function (err) {
               if (err) {
                 logger.info('Error when trying to execute the postLoginHandler after authenticating the user.');
                 return callback(err);
               }
 
-              callback(null, expandedAccount, authResult);
+              callback(null, user, oauthAccessTokenResult);
             });
           }
 
-          callback(null, expandedAccount, authResult);
+          callback(null, user, oauthAccessTokenResult);
```

Comment:
Ah, good catch, thanks!  I do need to mat this match the [https://docs.stormpath.com/nodejs/jsdoc/OAuthPasswordGrantAuthenticationResult.html](OAuthPasswordGrantAuthenticationResult) as much as possible.